sentry-core/src/main/java/io/sentry/transport/TransportResult.java

```diff
@@ -0,0 +1,85 @@
+package io.sentry.transport;
+
+import io.sentry.ISerializer;
+import io.sentry.SentryEvent;
+
+/**
+ * A result of {@link ITransport#send(SentryEvent, ISerializer)}. Note that this class is
+ * intentionally not subclassable and has only two factory methods to capture the 2 possible states
+ * - success or error.
+ */
+public abstract class TransportResult {
```

Comment:
Well, this class is not inheritable, because it has a private constructor (intentionally). It just a little bit of mostly unnecessary gymnastics, but with this approach, the "success" instance is a singleton and doesn't need to hold any data. The "error" results don't need to hold a boolean saying that they're not a success.

I can simplify this by doing what you suggest at the cost of ever so slightly higher memory consumed by the instances.

Reply:
The approach used by @metlos is the *"default"* way to get a sealed class in Java. I think it is fine.
It would have been nicer to read in Kotlin 😛

Reply:
yeah, I'd agree with @ninniuz that's the java verbose way of doing it.

Reply:
That's solves it then. Thanks everyone for chipping in